src/sync/sync_session.cpp

```diff
@@ -722,11 +722,13 @@ void SyncSession::add_completion_callback(_impl::SyncProgressNotifier::NotifierT
 {
     bool is_download = direction == _impl::SyncProgressNotifier::NotifierType::download;
 
-    int resync_counter = m_client_resync_counter;
-    std::weak_ptr<SyncSession> weak_self = shared_from_this();
     auto waiter = is_download ? &sync::Session::async_wait_for_download_completion
                               : &sync::Session::async_wait_for_upload_completion;
-    (m_session.get()->*waiter)([resync_counter, weak_self, is_download](std::error_code ec) {
+    (m_session.get()->*waiter)([is_download,
+                                resync_counter = m_client_resync_counter,
```

Comment:
I think the resync_counter logic is now redundant since if there was a resync, it won't find the entry for this id in the map. If you agree, I will remove that logic.

Reply:
Yes, this looks like a more general solution to the problem the resync counter was addressing.